Assignment 5_JUNG.Rmd

```diff
@@ -0,0 +1,107 @@
+
+#Visualize our outcome variable "score"
+```{r}
+hist(D1$score)
+```
+
+#Create a categorical outcome variable based on student score to advise the teacher using an "ifelse" statement
+```{r}
+D1$advice <- ifelse(D1$score<=0.4, "intervene", ifelse(D1$score>0.4 & D1$score<=0.8,"monitor","no action"))
+```
+
+#Build a decision tree that predicts "advice" based on how many problems students have answered before, the percentage of those problems they got correct and how many hints they required
+```{r}
+score_ctree <-ctree(factor(advice) ~ prior_prob_count + prior_percent_correct + hints, data=D1) 
+
+```
+
+#Plot tree
+```{r}
+plot(score_ctree)
+```
+
+Please interpret the tree, which two behaviors do you think the teacher should most closely pay attemtion to?
+# Students who do not ask for any hints (<=0) and have prior_percent_correct less than 85 (<=85): they are struggling but also do not have much motivation
+  ## only around 20% asks for intervention.
+# Students who do not ask for any hints (<=0) and have prior_percent_correct less than 85 (>85): they are getting the problems right and therefore do not need the intervention
+  ## Very samll portion (less than .1) are asking for intervention. Almost 70% of students in this group are asking for no action. 
+# Students who ask for hints but number of hints is less than or equal to 12 and have prior_percent_correct less than 63 (<=.629): they have some motivation, but are not doing so well
+  ## around 30% asks for intervention; 60% asks for motinoring; only 10% is asking for no action
+# Students who ask for hints but number of hints is greater than 12: they have a great motivation
+  ## less than 5% of them are asking for no action. High percentage of students are asking for either intervention or monitoring.
+  
+#Test Tree
+Upload the data "intelligent_tutor_new.csv". This is a data set of a differnt sample of students doing the same problems in the same system. We can use the tree we built for the previous data set to try to predict the "advice" we should give the teacher about these new students. 
+
+```{r}
+#Upload new data
+
+D2 <- read_csv("intelligent_tutor_new.csv")
+
+#Generate predicted advice using the predict() command for new students based on tree generated from old students
+
+D2$prediction <- predict(score_ctree,D2)
+
+``` 
+## Part III
+Compare the predicted advice with the actual advice that these students recieved. What is the difference between the observed and predicted results?
+```{r}
+Table <- table(D2$prediction)
+
+#monitor
+Table[2]/sum(Table)
+
+#no action
+Table[3]/sum(Table)
+
+
+
+```
+
+  #Score for everyone is 1 so there should "no actions" for all - and therefore, a lot of predictions are wrong. The model is not good.
```

Comment:
Try to calculate the accuracy rate and specify whether the model overfits or whether it is generalizable in terms of the accuracy and error rates. Nonetheless, good job overall!